pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -532,6 +557,68 @@ type PDSpec struct {
 	// Timeout threshold when pd get started
 	// +kubebuilder:default=30
 	StartTimeout int `json:"startTimeout,omitempty"`
+
+	// Mode is the mode of PD cluster
+	// +optional
+	// +kubebuilder:validation:Enum:="";"ms"
+	Mode string `json:"mode,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// PDMSSpec contains details of PD Micro Service
+type PDMSSpec struct {
+	ComponentSpec               `json:",inline"`
+	corev1.ResourceRequirements `json:",inline"`
+
+	// Name of the PD Micro Service
+	Name string `json:"name"`
```

Comment:
add a `+kubebuilder:validation:Enum` validation? and define the names in somewhere as consts

Reply:
got it